Equinox/Cube.cpp

```diff
@@ -18,73 +22,115 @@ Cube::~Cube() {}
 void Cube::Draw()
 {
 	glPushMatrix();
+
+	glTexEnvf(GL_TEXTURE_ENV, GL_TEXTURE_ENV_MODE, GL_DECAL);
+	glBindTexture(GL_TEXTURE_2D, _textureId);
 	glTranslatef(Position.x, Position.y, Position.z);
 	float3 axis = Rotation.Axis();
- 	glRotatef(RadToDeg(Rotation.Angle()), axis.x, axis.y, axis.z);
+	glRotatef(RadToDeg(Rotation.Angle()), axis.x, axis.y, axis.z);
 
 	glColor3f(Color.x, Color.y, Color.z);
 
 	glBegin(GL_TRIANGLES);
+	//a,b,d
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(0, 0, 0);
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(1, 0, 0);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(1, 1, 0);
 
 	//a,d,c
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(0, 0, 0);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(1, 1, 0);
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(0, 1, 0);
 
-	//b,f,h
+	//c,f,h
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(1, 0, 0);
+	glTexCoord2f(1.f, 0.f);
 	glVertex3f(1, 0, 1);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(1, 1, 1);
 
 	//b,h,d
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(1, 0, 0);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(1, 1, 1);
+	glTexCoord2f(1.f, 0.f);
 	glVertex3f(1, 1, 0);
 
 	//f,e,g
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(1, 0, 1);
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(0, 0, 1);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(0, 1, 1);
 
 	//f,g,h
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(1, 0, 1);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(0, 1, 1);
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(1, 1, 1);
 
 	//e,a,c
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(0, 0, 1);
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(0, 0, 0);
+	glTexCoord2f(1.f, 0.f);
 	glVertex3f(0, 1, 0);
 
-	//e,c,g
+	//e,b,g
+	glTexCoord2f(1.f, 0.f);
 	glVertex3f(0, 0, 1);
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(0, 1, 0);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(0, 1, 1);
 
 	//b,a,e
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(1, 0, 0);
+	glTexCoord2f(0.f, 0.f);
 	glVertex3f(0, 0, 0);
+	glTexCoord2f(1.f, 0.f);
 	glVertex3f(0, 0, 1);
 
-	//b,e,f
+	//c,e,f
+	glTexCoord2f(1.f, 0.f);
 	glVertex3f(1, 0, 0);
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(0, 0, 1);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(1, 0, 1);
 
 	//d,h,g
+	glTexCoord2f(1.f, 0.f);
 	glVertex3f(1, 1, 0);
+	glTexCoord2f(1.f, 1.f);
 	glVertex3f(1, 1, 1);
+	glTexCoord2f(0.f, 1.f);
 	glVertex3f(0, 1, 1);
 
-	//d,g,c
+	//d,g,b
```

Comment:
The comments was correct, b was (1, 0, 0) and c was (0, 1, 0), but I'm agree if you prefer change b to c, but then, change all other comments. Like that -> //e,a,c wil be //e,a,b